### PR TITLE
[sw/test_rom] Skip SRAM request for new key and LFSR init in test ROM.

### PR DESCRIPTION
**Pull request by timothytrippel**
The test `boot_rom` initialization code was requesting and new SRAM scrambing key and re-initializing SRAM to pseudorandom values in the ROM init asm. This wastes sim cycles as this is not production mask ROM, and is used solely for testing.

This commit removes this extra initialization, addressing a task in #9163.

Signed-off-by: Timothy Trippel <ttrippel@google.com>

### CODE REVIEW

**Comment by timothytrippel** [comment]
Marking this draft for now, until discussion during SW WG next week.

**Comment by mundaym** [comment]
You could also remove the register clobbering if you want (will only save a few cycles but is unnecessary for the test rom).

**Comment by timothytrippel** [comment]
> You could also remove the register clobbering if you want (will only save a few cycles but is unnecessary for the test rom).

Fixed.

File: sw/device/boot_rom/rom_crt.S [hunk 1]
```diff
@@ -94,49 +94,17 @@ _start:
...
-  //la   a1, _stack_end
-  //call crt_section_clear
-
   // Initialize the `.data` segment from the `.idata` segment.
   la   a0, _data_start
   la   a1, _data_end
   la   a2, _data_init_start
   call crt_section_copy
 
-  // Clobber all temporary registers.
```
Comment (tjaychen) [inline sw/device/boot_rom/rom_crt.S:121]: ah for DV purposes you MIGHT have to keep this.
It doesn't happen with all processors, but at least on some if you don't do this kind of clobbering, the "unknown values" can sometimes cause the simulation to get stuck.

**Review by tjaychen — APPROVED** [review]

File: sw/device/boot_rom/rom_crt.S (hunk 1 above)
Comment (sriyerg) [reply sw/device/boot_rom/rom_crt.S]: Curious - was that because on those processors, the registers were somehow not initialized to 0s on reset and they got stuck because of X-prop?

Comment (tjaychen) [reply sw/device/boot_rom/rom_crt.S]: yeah that's right. I think on those the registers were not reset at all, so the x's would just happily flow through address computations and eventually things would get stuck after a handful of operations. 

i think i've seen this on 3 separate arm processors.  I honestly don't remember if i ever saw it on ibex.

Comment (sriyerg) [reply sw/device/boot_rom/rom_crt.S]: Perhaps they used non-resettable flops for low power designs. On Ibex, the register file seems to be[ reset to 0](https://github.com/lowRISC/ibex/blob/1bbe27effeda63f34c4f5c06cc88da58f9c5a404/rtl/ibex_register_file_ff.sv#L58), so maybe this is ok?

**Comment by timothytrippel** [comment]
After consensus in SW WG, this PR is ready for review/merge (when all tests pass).

File: hw/top_englishbreakfast/util/sw_sources.patch [hunk 2]
```diff
@@ -111,9 +111,9 @@ index 0336455da..a46911701 100644
 -static dif_csrng_t csrng;
 -static dif_edn_t edn0;
 -static dif_edn_t edn1;
- 
+
  // TODO(alphan): Handle return values as long as they don't affect capture rate.
-
```
Comment (mundaym) [inline hw/top_englishbreakfast/util/sw_sources.patch:116]: There seems to be some unrelated whitespace changes being made to this patch. I noticed the same thing in a previous PR by @doreis-lowrisc too. Any idea what is causing it?

**Review by mundaym — APPROVED** [review]
